### ISSUE TITLE
[TechDocs CLI] Ability to auto generate an mkdocs.yml file dynamically

### ISSUE THREAD

**Issue by andrewthauer** [backstage/backstage#8107]
It would be nice to have the ability to publish docs sites that do not have an `mkdocs.yaml` file. This could be done by auto-generating the `mkdocs.yml` file based on a template and injecting some variables.

For example a template might looks something like this (using handlebars):

```yaml
site_name: {{siteName}}
site_description: {{siteDescription}}
site_url: {{siteUrl}}
repo_url: {{repoUrl}}

{{#if googleAnalytics}}
google_analytics:
  {{#each googleAnalytics}}
  - {{this}}
  {{/each}}
{{/if}}

plugins:
  - techdocs-core
{{#if plugins}}
  {{#each plugins}}
  - {{this}}
  {{/each}}
{{/if}}
```

This feature would create even less friction for teams adopting a `docs-like-code` approach. Also, this feature and some re-usable CI code, could make it possible to build a docs site automatically when simply detecting a `docs` directory in a repo.

This seems related to backstage/techdocs-cli#29 

NOTE: I started looking at doing this a while back before the `techdocs-cli` was extracted into it's own repo. The early beginnings of this can be found here - https://github.com/backstage/backstage/compare/master...andrewthauer:feat/techdocs-auto-config. Unfortunately, I have not had time to finish this.

**Comment by awanlin** [comment]
This ties it very much with issue #15413. @iamEAP can you re-open this so we can tackle both in PR #15417, please?

**Comment by andrewthauer** [comment]
I think this could possibly be solved in #15417 or an extension of it based on comments in #15413.
